src/main/java/com/salesforce/storm/spout/dynamic/coordinator/SpoutMonitor.java

```diff
@@ -209,6 +209,21 @@ public boolean hasSpout(final VirtualSpoutIdentifier virtualSpoutIdentifier) {
         }
     }
 
+    /**
+     * Signals to a VirtualSpout to stop, ultimately removing it from the monitor.
+     * @param virtualSpoutIdentifier identifier of the VirtualSpout instance to request stopped.
+     */
+    public void removeVirtualSpout(final VirtualSpoutIdentifier virtualSpoutIdentifier) {
+        if (!hasVirtualSpout(virtualSpoutIdentifier)) {
+            throw new IllegalArgumentException(
```

Comment:
probably should be consistent and toss the same exception as the other methods when the vspout id isn't found?